src/_P021_Level.ino

```diff
@@ -179,6 +203,24 @@ boolean Plugin_021(uint8_t function, struct EventStruct *event, String& string)
       success = true;
       break;
     }
+    case PLUGIN_ONCE_A_SECOND:
+    {
+      // When set but not Save after change is set, don't want to save twice
+      if ((P021_AUTOSAVE_TIMER > 0) && (P021_DONT_ALWAYS_SAVE != 0)) {
```

Comment:
Why not start a `PLUGIN_TASKTIMER_IN` ?
In there you can store some values, like the set value.
Then you can check if a new value is different from the one waiting to be saved so you don't need to reset the timer, etc.
Also it is a good idea to check if the final value differs from the currently stored one and then don't save ;)
So an accidentally changed and corrected level value is not saved.

Reply:
I didn't use the `PLUGIN_TASKTIMER_IN` as I don't expect that timer to survive a warm boot, and storing a time-out value in a UserVar does. Don't want to reset the timer, but want to keep the interval a sort of periodical, only starting it when the Set Level is actually changed. Standard issue with an offset-based timer like this that it won't work as intended when the ESP is put in sleep mode.

Adding a check for a change/revert would be a smart improvement indeed.

Reply:
> Adding a check for a change/revert would be a smart improvement indeed.

We had such a check in the past, when we still used the MD5 checksum.
This code is still present, but no longer used as we ran into issues when extending the settings file.
Maybe it is a good idea to re-activate this code and patch the checksum calculation by actually skipping the offset where the checksum is located.